torch/distributions/transformed_distribution.py

```diff
@@ -6,21 +6,28 @@
 
 class TransformedDistribution(Distribution):
     r"""
-    Extension of the Distribution class, which applies a sequence of Transforms to a base distribution.
-    Let f be the composition of transforms applied,
-    X ~ BaseDistribution
-    Y = f(X) ~ TransformedDistribution(BaseDistribution, f)
-    log p(Y) = log p(X) + log det (dX/dY)
+    Extension of the Distribution class, which applies a sequence of Transforms
+    to a base distribution.  Let f be the composition of transforms applied,
+
+        X ~ BaseDistribution
+        Y = f(X) ~ TransformedDistribution(BaseDistribution, f)
+        log p(Y) = log p(X) + log det (dX/dY)
     """
-    def __init__(self, base_distribution, transforms=[], *args, **kwargs):
-        super(TransformedDistribution, self).__init__(*args, **kwargs)
+    def __init__(self, base_distribution, transforms):
         self.base_dist = base_distribution
         if isinstance(transforms, Transform):
             self.transforms = [transforms, ]
         elif isinstance(transforms, list):
             if not all(isinstance(t, Transform) for t in transforms):
                 raise ValueError("transforms must be a Transform or a list of Transforms")
             self.transforms = transforms
+        else:
+            raise ValueError("transforms must be a Transform or list, but was {}".format(transforms))
+        shape = self.base_dist.batch_shape + self.base_dist.event_shape
+        event_dim = max([len(self.base_dist.event_shape)] + [t.event_dim for t in self.transforms])
+        batch_shape = shape[:len(shape) - event_dim]
+        event_shape = shape[len(shape) - event_dim:]
```

Comment:
I think these can just be `shape[:-event_dim]` and `shape[-event_dim:]`.

Reply:
Careful, negative indexing doesn't work when the argument is zero:
```py
shape = [1,2,3]
assert shape[-3:] == [1, 2, 3]
assert shape[-2:] == [2, 3]
assert shape[-1:] == [3]
assert shape[-0:] == [1, 2, 3]  # <---- this has tripped me up so often
```

Reply:
Got it; I figured there was probably a reason you were doing it that way. Thanks for letting me know!